File: Data Structures & Algorithms/two-integer-sum/submission-8.py

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        #input: a list of ints
        #output: is going to be the indices of the two numbers that hit target
        # Need to go through each number atleast once, 
        # Need to also keep track of indices some how
        # calculate what the target num should be
        # Constraints: can have negative numbers,
        mp = {}


        for i, num in enumerate(nums):
            t  = target - num
            if t in mp:
                return [mp[t], i]
            mp[num] = i
        return []

File: Data Structures & Algorithms/duplicate-integer/submission-8.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        const dups = new Set();

        for(let n of nums){
            if(dups.has(n)){
                return true
            }
            else{
                dups.add(n)
            }
        }
        return false;





    }
}
